pymc/tests/test_shape_handling.py

```diff
@@ -249,69 +247,60 @@ def test_param_and_batch_shape_combos(
         with pm.Model() as pmodel:
             x = pm.ConstantData("x", [1, 2, 3], dims="ddata")
 
-            if ellipsis_in == "none":
-                # The shape and dims tuples correspond to each other.
-                # Note: No checks are performed that implied shape (x), shape and dims actually match.
-                y = pm.Normal("y", mu=x, shape=(2, 3), dims=("dshape", "ddata"))
-                assert pmodel.RV_dims["y"] == ("dshape", "ddata")
-            elif ellipsis_in == "shape":
-                y = pm.Normal("y", mu=x, shape=(2, ...), dims=("dshape", "ddata"))
-                assert pmodel.RV_dims["y"] == ("dshape", "ddata")
-            elif ellipsis_in == "dims":
-                y = pm.Normal("y", mu=x, shape=(2, 3), dims=("dshape", ...))
-                assert pmodel.RV_dims["y"] == ("dshape", None)
-            elif ellipsis_in == "both":
-                y = pm.Normal("y", mu=x, shape=(2, ...), dims=("dshape", ...))
-                assert pmodel.RV_dims["y"] == ("dshape", None)
+            # The shape and dims tuples correspond to each other.
+            # Note: No checks are performed that implied shape (x), shape and dims actually match.
+            y = pm.Normal("y", mu=x, shape=(2, 3), dims=("dshape", "ddata"))
+            assert pmodel.RV_dims["y"] == ("dshape", "ddata")
 
             assert "dshape" in pmodel.dim_lengths
             assert y.eval().shape == (2, 3)
 
-    @pytest.mark.parametrize("with_dims_ellipsis", [False, True])
-    def test_simultaneous_size_and_dims(self, with_dims_ellipsis):
+    def test_simultaneous_size_and_dims(self):
         with pm.Model() as pmodel:
             x = pm.ConstantData("x", [1, 2, 3], dims="ddata")
             assert "ddata" in pmodel.dim_lengths
 
             # Size does not include support dims, so this test must use a dist with support dims.
             kwargs = dict(name="y", size=(2, 3), mu=at.ones((3, 4)), cov=at.eye(4))
-            if with_dims_ellipsis:
-                y = pm.MvNormal(**kwargs, dims=("dsize", ...))
-                assert pmodel.RV_dims["y"] == ("dsize", None, None)
-            else:
-                y = pm.MvNormal(**kwargs, dims=("dsize", "ddata", "dsupport"))
-                assert pmodel.RV_dims["y"] == ("dsize", "ddata", "dsupport")
+            y = pm.MvNormal(**kwargs, dims=("dsize", "ddata", "dsupport"))
+            assert pmodel.RV_dims["y"] == ("dsize", "ddata", "dsupport")
 
             assert "dsize" in pmodel.dim_lengths
             assert y.eval().shape == (2, 3, 4)
 
+    def test_simultaneous_dims_and_observed(self):
+        with pm.Model() as pmodel:
+            x = pm.ConstantData("x", [1, 2, 3], dims="ddata")
+            assert "ddata" in pmodel.dim_lengths
+
+            # Note: No checks are performed that observed and dims actually match.
+            y = pm.Normal("y", observed=[0, 0, 0], dims="ddata")
+            assert pmodel.RV_dims["y"] == ("ddata",)
+            assert y.eval().shape == (3,)
```

Comment:
Don't you want to broadcast in this test too?
```suggestion
            y = pm.Normal("y", observed=[[0, 0, 0]], dims=("dobs", "ddata"))
            assert pmodel.RV_dims["y"] == ("dobs", "ddata")
            assert y.eval().shape == (1, 3)
```

Reply:
Not sure, is this testing something that is not covered by `test_define_dims_on_the_fly_from_observed`? If it is the same I would like to keep it separate, in case we decide to drop support for dims on the fly.